### ISSUE TITLE
Error when restarting after editing /etc/zulip/zulip.conf

### ISSUE THREAD

**Issue by kevvvinyao** [zulip/docker-zulip#399]
I installed zulip via **docker**.
I want to put my zulip application behind a reverse proxy, following the document on https://zulip.readthedocs.io/en/stable/production/deployment.html#putting-the-zulip-application-behind-a-reverse-proxy 
Therefore, I edited  `/etc/zulip/zulip.conf` according to the tutorial. Then I run `/home/zulip/deployments/current/scripts/zulip-puppet-apply` and input `y` so that the changes will be applied.
However, when I wanted to **restart** the zulip server, error occured.
Below is my command to **restart** the zulip server:
``` shell
su zulip -c '/home/zulip/deployments/current/scripts/restart-server'
```
Below is output:
```
root@7ca519b88979:~# su zulip -c '/home/zulip/deployments/current/scripts/restart-server'
2023-03-30 13:41:29,621 restart-server: Running syntax and database checks
Traceback (most recent call last):
  File "./manage.py", line 145, in <module>
    log_management_command(sys.argv, settings.MANAGEMENT_LOG_PATH)
  File "/srv/zulip-venv-cache/b2cdd20dd55ad18cd1e512082a3ec530e6b6ece9/zulip-py3-venv/lib/python3.8/site-packages/django/conf/__init__.py", line 92, in __getattr__
    self._setup(name)
  File "/srv/zulip-venv-cache/b2cdd20dd55ad18cd1e512082a3ec530e6b6ece9/zulip-py3-venv/lib/python3.8/site-packages/django/conf/__init__.py", line 79, in _setup
    self._wrapped = Settings(settings_module)
  File "/srv/zulip-venv-cache/b2cdd20dd55ad18cd1e512082a3ec530e6b6ece9/zulip-py3-venv/lib/python3.8/site-packages/django/conf/__init__.py", line 190, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/zulip/deployments/2023-01-23-18-51-59/zproject/settings.py", line 27, in <module>
    from .computed_settings import *  # noqa: F403 isort: skip
  File "/home/zulip/deployments/2023-01-23-18-51-59/zproject/computed_settings.py", line 75, in <module>
    SECRET_KEY = get_mandatory_secret("secret_key")
  File "/home/zulip/deployments/2023-01-23-18-51-59/zproject/config.py", line 50, in get_mandatory_secret
    raise ZulipSettingsError(f'Mandatory secret "{key}" is not set')
zproject.config.ZulipSettingsError: Mandatory secret "secret_key" is not set
Traceback (most recent call last):
  File "/home/zulip/deployments/current/scripts/restart-server", line 47, in <module>
    subprocess.check_call(["./manage.py", "check", "--database", "default"])
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['./manage.py', 'check', '--database', 'default']' returned non-zero exit status 1.
```
I do not understand at all.
Then I tried re-install and start all over again, finding that: If I do not edit `/etc/zulip/zulip.conf`, the **restart** command could be run normally; If I edit `/etc/zulip/zulip.conf`, the **restart** command output error logs above.
I have no idea about why that happened.

**Comment by keithdanielll** [comment]
Same here after I edited the emailsettings according to https://zulip.readthedocs.io/en/latest/production/email.html

**Comment by fyrfli** [comment]
same here ... editing to add in the allowed ips though (i am running behind a reverse proxy).

**Comment by peracchi** [comment]
Same problem.

**Comment by timabbott** [comment]
Generally the Docker container expects you to not manually edit files in `/etc/`. If you're using the environment variables, see:

https://github.com/zulip/docker-zulip#configuration

I did earlier today improve the documentation to make it more obvious how to configure a reverse proxy there. Closing this. But folks encountering similar issues should ask for help in https://zulip.com/development-community/; that's going to be a lot more timely than posting in the issue tracker a configuration issue like this that cannot be debugged without the user present.
